src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs

```diff
@@ -289,6 +296,64 @@ public IActionResult GetUtxoSet(int atBlockHeight)
             }
         }
 
+        /// <summary>Provides verbose balance data of the given addresses.</summary>
+        /// <param name="address">The address to query for unspent outputs.</param>
+        /// <returns>A result object containing the UTXOs.</returns>
+        /// <response code="200">Returns the UTXO set for a particular address at the current chain tip height.</response>
+        /// <response code="400">Unexpected exception occurred</response>
+        [Route(BlockStoreRouteEndPoint.GetUtxoSetForAddress)]
+        [HttpGet]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public IActionResult GetUtxoSetForAddress(string address)
+        {
+            // Get coinview at current height (SLOW)
+            var coinView = this.utxoIndexer.GetCoinviewAtHeight(this.chainIndexer.Height);
+
+            var utxos = new List<UtxoModel>();
+
+            // Get utxos for this address
+            try
+            {
+                // Fine to do this in parallel because we don't care about the order
+                Parallel.ForEach(coinView.UnspentOutputs, (utxo) =>
```

Comment:
Would it be possible/more performance to break this loop when the amount needed for a transaction is found? 

Example: I need uxtos to cover 50 CRS, only return UXTOs needed for up to that amount (maybe plus 1 for gas).

Maybe not an option or worth it, just a thought. Most users of this endpoint will only need _enough_ to cover the transaction not necessary _all_ of them.

Reply:
Definitely. That sounds like a UTXO selection endpoint which isn't the goal with this PR, but could be added after the utxo indexer has been optimized.